### ISSUE TITLE
Run build.sh on branch `v2.1.4` doesn't work on Linux

### ISSUE THREAD

**Issue by dgodd** [dotnet/cli#8358]
## Steps to reproduce
docker run  -it debian:8 bash

Note: I used debian:8 due to https://github.com/dotnet/cli/issues/8250

```
apt-get update
apt-get install git curl libunwind-dev libicu-dev

git clone https://github.com/dotnet/cli
cd cli/
git checkout v2.1.4
export DropSuffix=true
./build.sh /t:Compile
```

## Expected  behavior
A built binary

## Actual behavior
I receive the following error

```
Host info - Rid: debian.8-x64, OSName: debian
Build info - Rid: linux-x64, Architecture: x64, OSName: linux
If you intended to use a different Rid, Architecture, or OSName run the following command to generate your build state:
dotnet msbuild build.proj /p:Architecture=Architecture /p:Rid=Rid /p:OSName=OSName /p:GeneratePropsFile=true /t:WriteDynamicPropsToStaticPropsFiles
2.0.3-servicing-007156
/cli/build/Prepare.targets(35,5): error MSB4018: The "DownloadFile" task failed unexpectedly. [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018: System.AggregateException: One or more errors occurred. (Response status code does not indicate success: 404 (Not Found).) ---> System.Net.Http.HttpRequestException: Response status code does not indicate success: 404 (Not Found). [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018:    at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode() [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018:    at System.Net.Http.HttpClient.<FinishGetStreamAsync>d__33.MoveNext() [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018:    --- End of inner exception stack trace --- [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018:    at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification) [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018:    at Microsoft.DotNet.Cli.Build.DownloadFile.Execute() in /cli/build_projects/dotnet-cli-build/DownloadFile.cs:line 57 [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext() [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018: ---> (Inner Exception #0) System.Net.Http.HttpRequestException: Response status code does not indicate success: 404 (Not Found). [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018:    at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode() [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018:    at System.Net.Http.HttpClient.<FinishGetStreamAsync>d__33.MoveNext()<--- [/cli/build.proj]
/cli/build/Prepare.targets(35,5): error MSB4018:  [/cli/build.proj]
```

## Environment data
`dotnet --info` output:

```
bash: dotnet: command not found
```

**Comment by livarcocc** [comment]
@johnbeisner can you take a look, since this is a install script failure?

**Comment by johnbeisner** [comment]
@dgodd 
This is indeed a regression in the scripts - thank you for pointing it out: https://github.com/dotnet/cli/issues/8375

Here are your options as I see it:
1) do not pass in a 'target'
2) Change line 191 - removing "${args[@]}"

From: dotnet msbuild build.proj /p:Architecture=$ARCHITECTURE $CUSTOM_BUILD_ARGS /p:GeneratePropsFile=true /t:WriteDynamicPropsToStaticPropsFiles "${args[@]}"
To: dotnet msbuild build.proj /p:Architecture=$ARCHITECTURE $CUSTOM_BUILD_ARGS /p:GeneratePropsFile=true /t:WriteDynamicPropsToStaticPropsFiles

**Comment by livarcocc** [comment]
@johnbeisner I believe you are working on fixing this as well, right?

**Comment by dgodd** [comment]
Thank you @johnbeisner , removing the "/t:Compile" appears to do the trick for us

**Comment by johnbeisner** [comment]
When passing in a 'target' on the command line from Build.sh/cmd; the target is attempted on the first pass call to "dotnet msbuild build.proj". This causes unpredictable results. PR: https://github.com/dotnet/cli/pull/8380

**Comment by johnbeisner** [comment]
@dgodd 
We are fixing this for future CLI releases.

**Comment by tylerphelan** [comment]
@johnbeisner @livarcocc We tried this again with cli v2.1.101
```
apt-get update
apt-get install git curl libunwind-dev libicu-dev

git clone https://github.com/dotnet/cli
cd cli/
git checkout v2.1.101
export DropSuffix=true
./build.sh /t:Compile
``` 
and are seeing the same error. Is there any update on this?

**Comment by johnbeisner** [comment]
@tylerphelan 
Unfortunately, the fix did not make it into cli v2.1.101; please remove the target.
